homeassistant/components/fan/dyson.py

```diff
@@ -10,62 +10,148 @@
 import homeassistant.helpers.config_validation as cv
 from homeassistant.components.dyson import DYSON_DEVICES
 from homeassistant.components.fan import (
-    DOMAIN, SUPPORT_OSCILLATE, SUPPORT_SET_SPEED, FanEntity)
-from homeassistant.const import CONF_ENTITY_ID
+    DOMAIN, SUPPORT_OSCILLATE, SUPPORT_SET_SPEED, FanEntity,
+    SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH)
+from homeassistant.const import ATTR_ENTITY_ID
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_NIGHT_MODE = 'night_mode'
-
-ATTR_IS_NIGHT_MODE = 'is_night_mode'
-ATTR_IS_AUTO_MODE = 'is_auto_mode'
+ATTR_NIGHT_MODE = 'night_mode'
+ATTR_AUTO_MODE = 'auto_mode'
+ATTR_ANGLE_LOW = 'angle_low'
+ATTR_ANGLE_HIGH = 'angle_high'
+ATTR_FLOW_DIRECTION_FRONT = 'flow_direction_front'
+ATTR_TIMER = 'timer'
+ATTR_DYSON_SPEED = 'dyson_speed'
+ATTR_DYSON_SPEED_LIST = 'dyson_speed_list'
 
 DEPENDENCIES = ['dyson']
 DYSON_FAN_DEVICES = 'dyson_fan_devices'
 
 SERVICE_SET_NIGHT_MODE = 'dyson_set_night_mode'
+SERVICE_SET_AUTO_MODE = 'dyson_set_auto_mode'
+SERVICE_SET_ANGLE = 'dyson_set_angle'
+SERVICE_SET_FLOW_DIRECTION_FRONT = 'dyson_flow_direction_front'
+SERVICE_SET_TIMER = 'dyson_set_timer'
+SERVICE_SET_DYSON_SPEED = 'dyson_set_speed'
 
 DYSON_SET_NIGHT_MODE_SCHEMA = vol.Schema({
-    vol.Required(CONF_ENTITY_ID): cv.entity_id,
-    vol.Required(CONF_NIGHT_MODE): cv.boolean,
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_NIGHT_MODE): cv.boolean,
+})
+
+DYSON_SET_AUTO_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_AUTO_MODE): cv.boolean,
+})
+
+DYSON_SET_ANGLE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_ANGLE_LOW): cv.positive_int,
```

Comment:
I think that its a bit of unfortunate naming here, since the method to enable oscillation and to set the angle is the same, so when called without any angles as parameters it will used the previously stored values for the angles and just enable the oscillation. Here there is a separate service to turn the oscillation on so it doesn't make much sense to call the service without any arguments. I'll change both to required.